src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/Util.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.SQLite
+{
+    internal static class Util
+    {
+        internal static byte[] ToUtf8(string text)
+        {
+            if (text == null)
+            {
+                return null;
+            }
+
+            // SQLite expects null terminated UTF8, so do conversion by allocating
```

Comment:
💭 This method could be reduced to the following if you want it simple:

```
return Encoding.UTF8.GetBytes(text + '\0');
```

Reply:
`Encoding.UTF8.GetBytes(string)` internally calls the byte[] variant, https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Text/Encoding.cs#L770-L781

So, that would incur:
1 string allocation '\0' -> "0"
1 string allocation for the string concat (text + "0")

I used this method for performance.

Reply:
> I used this method for performance.

Relative impact of costs needs to be considered.  So, for example, your change helps replace an unending amount of string->byte[] allocations with a single fixed number of allocations (<10).  Improving allocation scenarios doesn't mean removing all allocations, just the costly ones :)

Reply:
that said, i'm fine with the code as is.  It's nice and clear and doesn't do anything that seems unnecessary.  Seems like a good solution to me.

Reply:
Ah, I just realized that I changed this locally before I moved the things to happen in the static ctors. I can simplify the code.